src/date-range-picker/calendar/index.tsx

```diff
@@ -42,62 +42,24 @@ export default function DateRangePickerCalendar({
   timeInputFormat = 'hh:mm:ss',
 }: DateRangePickerCalendarProps) {
   const isSingleGrid = useMobile();
-
   const normalizedLocale = normalizeLocale('DateRangePicker', locale);
   const normalizedStartOfWeek = normalizeStartOfWeek(startOfWeek, normalizedLocale);
 
-  const initialStartDate = value?.startDate ?? '';
-  const rangeStart = useDateTime(initialStartDate);
-
-  const initialEndDate = value?.endDate ?? '';
-  const rangeEnd = useDateTime(initialEndDate);
-
   const [announcement, setAnnouncement] = useState('');
-
-  const [currentMonth, setCurrentMonth] = useState(() => {
-    if (rangeStart.dateString) {
-      const startDate = parseDate(rangeStart.dateString);
-      if (isSingleGrid) {
-        return startOfMonth(startDate);
-      }
-      return startOfMonth(addMonths(startDate, 1));
-    }
-    if (rangeEnd.dateString) {
-      return startOfMonth(parseDate(rangeEnd.dateString));
-    }
-    return startOfMonth(Date.now());
-  });
-
+  const [currentMonth, setCurrentMonth] = useState(() => findMonthToDisplay(value, isSingleGrid));
   const [focusedDate, setFocusedDate] = useState<Date | null>(() => {
-    if (rangeStart.date) {
-      if (isSameMonth(rangeStart.date, currentMonth)) {
-        return rangeStart.date;
+    if (value.start.date) {
+      const startDate = parseDate(value.start.date);
+      if (isSameMonth(startDate, currentMonth)) {
+        return startDate;
       }
-      if (!isSingleGrid && isSameMonth(rangeStart.date, addMonths(currentMonth, -1))) {
-        return rangeStart.date;
+      if (!isSingleGrid && isSameMonth(startDate, addMonths(currentMonth, -1))) {
+        return startDate;
       }
     }
-    return selectFocusedDate(rangeStart.date, currentMonth, isDateEnabled);
+    return findDateToFocus(parseDate(value.start.date), currentMonth, isDateEnabled);
   });
 
-  // This effect "synchronizes" the local state update back up to the parent component.
-  useEffect(() => {
```

Comment:
React docs explicitly call this an anti-pattern: https://beta.reactjs.org/learn/you-might-not-need-an-effect#notifying-parent-components-about-state-changes

Avoid code patterns like this